javalib/src/main/scala/java/lang/Long.scala

```diff
@@ -338,9 +338,82 @@ object Long {
   @inline def valueOf(s: String, radix: Int): Long =
     valueOf(parseLong(s, radix))
 
-  // TODO:
-  // def parseUnsignedLong(s: String): scala.Long = parseUnsignedLong(s, 10)
-  // def parseUnsignedLong(s: String, radix: Int): scala.Long = ???
-  // def toUnsignedString(l: scala.Long): String = toUnsignedString(l, 10)
-  // def toUnsignedString(l: scala.Long, radix: Int): String = ???
+  @inline def parseUnsignedLong(s: String): scala.Long =
+    parseUnsignedLong(s, 10)
+
+  def parseUnsignedLong(s: String, radix: Int): scala.Long = {
+    if (s == null || radix < Character.MIN_RADIX ||
+        radix > Character.MAX_RADIX) throw new NumberFormatException(s)
+
+    val len = s.length()
+
+    if (len == 0) throw new NumberFormatException(s)
+
+    val hasPlusSign = s.charAt(0) == '+'
+
+    if (hasPlusSign && len == 1) throw new NumberFormatException(s)
+
+    val offset = if (hasPlusSign) 1 else 0
+
+    parseUnsigned(s, offset, radix)
+  }
+
+  private def parseUnsigned(s: String, _offset: Int, radix: Int): scala.Long = {
+    val unsignedLongMaxValue = -1L
+    val max                  = divideUnsigned(unsignedLongMaxValue, radix)
+    var result = 0L
+    var offset = _offset
+    val length = s.length()
+
+    while (offset < length) {
+      val digit = Character.digit(s.charAt(offset), radix)
+      offset += 1
+
+      if (digit == -1) throw new NumberFormatException(s)
+
+      if (compareUnsigned(result, max) == 1) throw new NumberFormatException(s)
+
+      result = result * radix + digit
+    }
```

Comment:
The same problem with addition overflow appears here. Test case: `JLong.parseUnsignedLong("18446744073709551616")`